examples/research_projects/codeparrot/scripts/human_eval_mgpu.py

```diff
@@ -0,0 +1,171 @@
+import json
+import multiprocessing
+import os
+import re
+
+from datasets import load_dataset, load_metric
+from tqdm import tqdm
+import torch
+from torch.utils.data import IterableDataset
+from torch.utils.data.dataloader import DataLoader
+
+from accelerate import Accelerator
+import transformers
+from arguments import HumanEvalArguments
+from transformers import (
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    HfArgumentParser,
+    StoppingCriteria,
+    StoppingCriteriaList,
+    set_seed,
+)
+
+
+EOF_STRINGS = ["\nclass", "\ndef", "\n#", "\n@", "\nprint", "\nif"]
+
+
+class TokenizeDataset(IterableDataset):
+    def __init__(self, tokenizer, dataset, max_length, n_tasks=None):
+        self.tokenizer = tokenizer
+        self.dataset = dataset
+        self.max_length = max_length
+        self.n_tasks = len(dataset) if n_tasks is None else n_tasks
+
+    def __iter__(self):
+        for task in range(self.n_tasks):
+            # why we need eos token? do we need strip?
```

Comment:
Very tricky .... indeed the model work in a wired way with strip ...
and the eos token is tricky as well

Reply:
Oh actually didn't see the question about strip: The issue is that the tokenizer has a token `\n\t` and the model learns that after a function signature this token comes. However, in the HumanEval dataset by default there is just a `\n` and although the model also has a `\t` token it has never seen a `\n` + `\t` during pretraining since that is always replaced by the single `\n\t` token. Therefore this would be a very uncommon sequence and results in a weird behaviour in the model.